vitest.config.ts

```diff
@@ -1,14 +1,21 @@
-import { defineConfig } from 'vitest/config';
+import { defineConfig, UserConfigExport } from 'vitest/config';
 
-export default defineConfig({
-  test: {
-    coverage: {
-      provider: 'istanbul',
-      lines: 0,
-      functions: 0,
-      statements: 0,
-      all: true,
+export const getVitestConfig = (fullyCoveredModulePaths?: string[]): UserConfigExport => {
+  const enableFullCoverage = !!fullyCoveredModulePaths && fullyCoveredModulePaths.length !== 0;
```

Comment:
Nit: Not sure why you've decided to complicate this and split the logic into two files instead of just exporting the pure config object here and overwriting its parts in the other file

Otherwise, you can also read the cli flag here and have a single config file